dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionHelper.java

```diff
@@ -87,18 +87,24 @@ public void evictAndAddEntries(Collection collection) {
             entry.setLabels(labelStrings);
             List<CategorySummary> summaries = entryDAO.findCategorySummariesByEntryId(entry.getId());
             entry.setCategorySummaries(summaries);
-            if (entry.getEntryType().equals("tool")) {
+            switch (entry.getEntryType()) {
+            case "tool":
                 entry.setDescriptorTypes(entryDAO.getToolsDescriptorTypes(entry.getId()));
-            } else if (entry.getEntryType().equals("workflow")) {
+                break;
+            case "workflow":
                 entry.setDescriptorTypes(entryDAO.getWorkflowsDescriptorTypes(entry.getId()));
+                break;
+            case "apptool":
+                entry.setDescriptorTypes(entryDAO.getWorkflowsDescriptorTypes(entry.getId()));
+                // we get file descriptor types like workflows, but make the UI treat these as workflows (so icon and url work)
```

Comment:
"but make the UI treat these as **tools**"

Reply:
(ninja-ed)